app/services/package.js

```diff
@@ -7,85 +7,112 @@ var registry = require('datapackage-registry');
 var utils = require('./utils');
 require('isomorphic-fetch');
 
-function getResourceName(resource, index) {
-  return resource.name || 'resource_' + index;
-}
-
-function FiscalDataPackage() {
-  this.attributes = {};
-  this.resources = [];
-
-  this.resources.add = function(resource) {
-    resource.name = utils.createUniqueName(
-      getResourceName(resource, this.length), _.pluck(this, 'name'));
-    this.push(resource);
-  };
+module.exports.createResourceFromSource = function(urlOrFile) {
+  return new Promise(function(resolve, reject) {
+    utils.getCsvSchema(urlOrFile)
+      .then(function(data) {
+        var resourceName = null;
+        var source = {};
+        if (_.isObject(urlOrFile)) {
+          source.fileName = urlOrFile.name;
+          source.mimeType = urlOrFile.type;
+          source.size = urlOrFile.size;
+          resourceName = utils.createNameFromPath(urlOrFile.name);
+        } else {
+          source.url = utils.undecorateProxyUrl(urlOrFile);
+          resourceName = utils.createNameFromUrl(urlOrFile);
+        }
 
-  this.resources.clear = function() {
-    this.splice(0, this.length);
-  };
+        var dataColumns = _.unzip(data.rows || []);
+
+        var resource = {
+          name: resourceName,
+          title: utils.convertToTitle(resourceName),
+          source: source,
+          data: {
+            headers: data.headers,
+            rows: data.rows,
+            raw: data.raw
+          },
+          fields: _.map(data.schema.fields, function(field, index) {
+            field = _.clone(field);
+            field.concept = (field.concept || '') + '';
+            field.inferredType = field.type;
+            field.title = utils.convertToTitle(field.name);
+            field.allowedTypes = utils.getAllowedTypesForValues(
+              dataColumns[index], field.inferredType);
+            field.allowedConcepts = utils.getAllowedConcepts(
+              field.allowedTypes);
+            return field;
+          })
+        };
+
+        resolve(resource);
+        return data;
+      })
+      .catch(reject);
+  });
+};
 
-  this.resources.createFromSource = function(urlOrFile) {
-    return new Promise(function(resolve, reject) {
-      utils.getCsvSchema(urlOrFile)
+module.exports.getFiscalDataPackageSchema = function() {
+  return module.exports.getDataPackageSchema('fiscal');
+};
+
+module.exports.getDataPackageSchema = function(schemaId) {
+  return new Promise(function(resolve, reject) {
+    var options = {
+      backend: '/proxy?url=' + encodeURIComponent('https://rawgit.com/' +
```

Comment:
* FYI this is now the offical location for the schema registry: http://schemas.datapackages.org/registry.csv